certora/ArrangerConduit.spec

```diff
@@ -0,0 +1,805 @@
+
+    assert balanceOfBufferAfter == balanceOfBufferBefore - amount, "deposit did not decrease balance of buffer by amount";
+    assert balanceOfConduitAfter == balanceOfConduitBefore + amount, "deposit did not increase balance of conduit by amount";
+    assert depositsAfter == depositsBefore + amount, "deposit did not increase deposits by amount";
+    assert totalDepositsAfter == totalDepositsBefore + amount, "deposit did not increase totalDeposits by amount";
+
+    assert depositsOtherAfter == depositsOtherBefore, "other deposits changed unexpectedly";
+    assert totalDepositsOtherAfter == totalDepositsOtherBefore, "other total deposits changed unexpectedly";
+}
+
+// Verify revert rules on deposit
+rule deposit_revert(bytes32 ilk, address asset, uint256 amount) {
+    env e;
+
+    require asset == gem;
+
+    address buffer = registry.buffers(ilk);
+    require buffer != currentContract;
+
+    bool canCall = roles.canCall(ilk, e.msg.sender, currentContract, to_bytes4(0xd954863c)); // deposit(bytes32,address,uint256)
+    mathint balanceOfBuffer = gem.balanceOf(buffer);
+    mathint allowanceBuffer = gem.allowance(buffer, currentContract);
+    mathint deposits = deposits(asset, ilk);
+    mathint totalDeposits = totalDeposits(asset);
+
+    deposit@withrevert(e, ilk, asset, amount);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = !canCall;
+    bool revert3 = deposits + amount > max_uint256;
+    bool revert4 = totalDeposits + amount > max_uint256;
+    bool revert5 = buffer == 0;
+    bool revert6 = balanceOfBuffer < to_mathint(amount);
+    bool revert7 = allowanceBuffer < to_mathint(amount);
+
+    assert lastReverted <=> revert1 || revert2 || revert3 ||
+                            revert4 || revert5 || revert6 ||
+                            revert7 , "Revert rules failed";
+
+}
+
+// Verify correct storage changes for non reverting withdraw
+rule withdraw(bytes32 ilk, address asset, uint256 maxAmount) {
+    env e;
+
+    require asset == gem;
+
+    address buffer = registry.buffers(ilk);
+    require buffer != currentContract;
+
+    mathint balanceOfBufferBefore = gem.balanceOf(buffer);
+    mathint balanceOfConduitBefore = gem.balanceOf(currentContract);
+    mathint withdrawableFundsBefore = withdrawableFunds(asset, ilk);
+    mathint totalWithdrawableFundsBefore = totalWithdrawableFunds(asset);
+    mathint withdrawalsBefore = withdrawals(asset, ilk);
+    mathint totalWithdrawalsBefore = totalWithdrawals(asset);
+
+    bytes32 otherIlk;
+    address otherAsset;
+    require otherIlk != ilk || otherAsset != asset;
+    mathint withdrawableFundsOtherBefore = withdrawableFunds(otherAsset, otherIlk);
+
+    address otherAsset2;
+    require otherAsset2 != asset;
+    mathint totalWithdrawableFundsOtherBefore = totalWithdrawableFunds(otherAsset2);
+
+    bytes32 otherIlk2;
+    address otherAsset3;
+    require otherIlk2 != ilk || otherAsset3 != asset;
```

Comment:
This is not wrong at all, but you could use the same `otherIlk` and `otherAsset` here, as they are tight to the same grouped key. As you just need to find a condition where this doesn't pass, it doesn't really matter if for each run both mappings are checked on the same key.
The problem to use `otherAsset` where you are using `otherAsset2` is that `otherAsset` could be equal to `asset` due to the `||` condition, and there you will have an unintended violation.
So basically, summarizing, you just need `otherIlk`, `otherAsset` and `otherAsset2` for this rule.